### PR TITLE
feat: network error handling for sso orchestrator self service

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
https://2u-internal.atlassian.net/browse/ENT-7617

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 1]
```diff
@@ -115,41 +116,48 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
...
+                    refreshBool={refreshBool}
+                    setRefreshBool={setRefreshBool}
+                  />
+                )}
+                {/* Nothing found so guide user to creation/edit form */}
+                {showNoSSOCard && <NoSSOCard setShowNoSSOCard={setShowNoSSOCard} setShowNewSSOForm={setShowNewSSOForm} />}
+                {/* Since we found a selected providerConfig we know we are in editing mode and can safely
+                render the create/edit form */}
+                {((existingConfigs?.length > 0 && providerConfig !== null) || showNewSSOForm) && (<NewSSOConfigForm />)}
+                {/* {error && (
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/index.jsx:137]: Not using this anymore?

**Review by marlonkeating — CHANGES_REQUESTED** [review]
I'm not seeing the list of SSO configs update after creating a new config.


https://github.com/openedx/frontend-app-admin-portal/assets/322346/5ddba2d0-fe3a-4314-94eb-9571c95e419a

File: src/components/settings/data/constants.js [hunk 2]
```diff
@@ -57,6 +57,9 @@ export const INVALID_ODATA_API_TIMEOUT_INTERVAL = 'OData API timeout interval mu
 
 export const MAX_UNIVERSAL_LINKS = 100;
 
+export const ssoStepperNetworkErrorText = `We were unable to configure your SSO due to an internal error.`;
+export const ssoLPNetworkErrorText = `We were unable to load your SSO details due to an internal error.`;
```
[the remark this answers is not in the document]
Comment (alex-sheehan-edx) [reply src/components/settings/data/constants.js:61]: I was following the examples of the other modal text [above](https://github.com/openedx/frontend-app-admin-portal/pull/1089/files/d8dec147f467950d50cbc031ad86d47c7d770bca#diff-0575f6750b5731d61fd3e2c8ca2df0bf5cf0454684c1419b226050497e11be0eR38-R40)

File: src/components/settings/SettingsSSOTab/index.jsx (hunk 1 above)
Comment (alex-sheehan-edx) [reply src/components/settings/SettingsSSOTab/index.jsx:137]: ah, yup! Good catch, it's just cruft

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 3]
```diff
@@ -115,41 +116,48 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
...
+                <Alert variant="warning" icon={WarningFilled}>
+                  An error occurred loading the SAML configs: <p>{error?.message}</p>
+                </Alert>
+                )} */}
+                {pdError && (
+                <Alert variant="warning" icon={WarningFilled}>
+                  An error occurred loading the SAML data: <p>{pdError?.message}</p>
+                </Alert>
+                )}
+                {infoMessage && (
```
[the remark this answers is not in the document]
Comment (alex-sheehan-edx) [reply src/components/settings/SettingsSSOTab/index.jsx:147]: interesting, this isn't changed from master, just the indentation moved with the error conditional. I'm not even sure what the info message is showing to the user.... let me look into it

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 4]
```diff
@@ -115,41 +117,44 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
...
+                render the create/edit form */}
+                {((existingConfigs?.length > 0 && providerConfig !== null) || showNewSSOForm) && (<NewSSOConfigForm />)}
+                {pdError && (
+                <Alert variant="warning" icon={WarningFilled}>
+                  An error occurred loading the SAML data: <p>{pdError?.message}</p>
+                </Alert>
+                )}
+                <Toast
+                  onClose={() => setInfoMessage(null)}
+                  show={infoMessage.length > 0}
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/index.jsx:148]: We're setting `infoMessage` to null when closing above.
```suggestion
                  show={infoMessage?.length > 0}
```

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/components/settings/SettingsSSOTab/SsoErrorPage.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  isOpen,
+  stepperError,
+}) => {
+  const stepperText = stepperError ? ssoStepperNetworkErrorText : ssoLPNetworkErrorText;
+  return (
+    <FullscreenModal
+      isOpen={isOpen}
+      variant="error"
+      hasCloseButton={false}
+      onClose={() => { }}
```
[the remark this answers is not in the document]
Comment (alex-sheehan-edx) [reply src/components/settings/SettingsSSOTab/SsoErrorPage.jsx:24]: UX has the modal as unclosable ¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (marlonkeating) [reply src/components/settings/SettingsSSOTab/SsoErrorPage.jsx:24]: Would omitting the `onClose` prop or setting it to `null` have the same effect?  IMO as it stands, it looks like we forgot to fill in some placeholder code.

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**